src/netCDF4/__init__.pyi

```diff
@@ -474,15 +615,14 @@ class Variable(Generic[T_Datatype]):
     def set_ncstring_attrs(self, ncstring_attrs: bool) -> None: ...
     def set_collective(self, value: bool) -> None: ...
     def get_dims(self) -> tuple[Dimension, ...]: ...
-
     def __delattr__(self, name: str) -> None: ...
     def __setattr__(self, name: str, value: Any) -> None: ...
     def __getattr__(self, name: str) -> Any: ...
-    def __getitem__(self, elem: GetSetItemKey) -> np.ndarray: ...
+    def __getitem__(self, elem: GetSetItemKey) -> Any: ...
     def __setitem__(self, elem: GetSetItemKey, data: npt.ArrayLike) -> None: ...
     def __array__(self) -> np.ndarray: ...
     def __len__(self) -> int: ...
-
+    def __iter__(self) -> Iterator[Any]: ...  # faux method so mypy believes Variable is iterable
```

Comment:
Can we specify the type a bit better here?

Reply:
Not easily, partly for the same reasoning as `__getitem__`, below. If the variable is 1-D, the iterator may return either a scalar or an array, depending on `auto_mask` (unless the variable is `str` type in which case it will always return a `str`). If it's multi-dimensional, it will return an array.